### PR TITLE
gnushogi: fix build

### LINKED ISSUE #211433: gnushogi build failed silently, leaving an empty output directory
### Steps To Reproduce

Position: https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/gnushogi/default.nix

Reproducible on the master branch as well as on nixos-22.11.

1.   build `gnushogi`
2.   `ls result/`

### Build log

| branch | date | commit | file | status |
|-|-|-|-|-|
| nixos-unstable | 2023-01-15 | 6dccdc458512abce8d19f74195bb20fdb067df50 | [ivy7byryh7als947x91cmqflawccjf5n-gnushogi-1.4.2.drv.log](https://gist.github.com/ShamrockLee/5af82885c5d00e99a6df9e089e552c71#file-ivy7byryh7als947x91cmqflawccjf5n-gnushogi-1-4-2-drv-log) | silent failure |
| nixos-22.11 | 2023-01-16 | 2f9fd351ec37f5d479556cd48be4ca340da59b8f | [avyysmzc3zxfi86x2k2a5v6dr93ysa39-gnushogi-1.4.2.drv.log](https://gist.github.com/ShamrockLee/5af82885c5d00e99a6df9e089e552c71#file-avyysmzc3zxfi86x2k2a5v6dr93ysa39-gnushogi-1-4-2-drv-log) | silent failure |
| nixos-21.11 | 2020-08-02 | eabc38219184cc3e04a974fe31857d8e0eac098d | [5akqnaxjr979zljs99h35zpg714fkc4v-gnushogi-1.4.2.drv.log](https://gist.github.com/ShamrockLee/5af82885c5d00e99a6df9e089e552c71#file-5akqnaxjr979zljs99h35zpg714fkc4v-gnushogi-1-4-2-drv-log) | success |

https://gist.github.com/ShamrockLee/5af82885c5d00e99a6df9e089e552c71

### Notify maintainers

@ciil

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.86, NixOS, 22.11 (Raccoon), 22.11.20230110.54644f4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/store/qwgw2b19ga8dr7c6wfpd6p87bdjyrnha-source`
```

### PR DESCRIPTION
**Pull request by nevivurn**
###### Description of changes

- Fix gnushogi build silently succeeding with empty output
- Example build: https://hydra.nixos.org/build/219348389/nixlog/1

Fixes #211433

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please mark broken on darwin

```

this derivation will be built:
  /nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv
this path will be fetched (0.37 MiB download, 0.37 MiB unpacked):
  /nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz
copying path '/nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz' from 'https://cache.nixos.org'...
building '/nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv'...
unpacking sources
unpacking source archive /nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz
source root is gnushogi-1.4.2
setting SOURCE_DATE_EPOCH to timestamp 1392669019 of file gnushogi-1.4.2/win32/Makefile
patching sources
configuring
configure flags: --prefix=/nix/store/y3197j3amzgca4549780w9vdv4i58idw-gnushogi-1.4.2
configure: C compiler
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/private/tmp/nix-build-gnushogi-1.4.2.drv-0/gnushogi-1.4.2':
configure: error: C compiler cannot create executables
See `config.log' for more details
error: builder for '/nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv' failed with exit code 77;
```

File: pkgs/games/gnushogi/default.nix [hunk 1]
```diff
@@ -10,6 +10,13 @@ stdenv.mkDerivation rec {
     sha256 = "0a9bsl2nbnb138lq0h14jfc5xvz7hpb2bcsj4mjn6g1hcsl4ik0y";
   };
 
+  env.LDFLAGS = "-Wl,-z,muldefs";
```
Comment (wegank) [inline pkgs/games/gnushogi/default.nix:13]:
```suggestion
  env.LDFLAGS = lib.optionalString (!stdenv.isDarwin) "-Wl,-z,muldefs";
```

**Review by wegank — APPROVED** [review]

File: pkgs/games/gnushogi/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/games/gnushogi/default.nix]: Does that fix darwin? If so please remove the mark broken again.

Comment (nevivurn) [reply pkgs/games/gnushogi/default.nix]: No, this doesn't fix darwin, I couldn't find an alternative for `-z muldefs` for clang from searching.

**Pull request merged**